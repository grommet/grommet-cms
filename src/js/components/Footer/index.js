/* @flow */
import React from 'react';
import FooterComponent from 'grommet/components/Footer';
import Box from 'grommet/components/Box';
import Heading from 'grommet/components/Heading';
import Anchor from 'grommet/components/Anchor';

export default function Footer(props: {
  logo: HTMLElement | React$Element<any>,
  contact: {
    copyright: string,
    email: string,
    website: string
  }
}) {
  const { logo, contact } = props;
  return (
    <FooterComponent primary colorIndex="brand">
      <Box direction="row" pad="large" full="horizontal">
        <Box basis="1/3" justify="center">
          {logo}
        </Box>
        <Box basis="1/3" justify="center">
          <Heading 
            className="post-feed-item--header"
            align="center"
            tag="h3"
          >
            {contact.copyright}
          </Heading>
        </Box>
        <Box basis="1/3" justify="center">
          <Anchor align="end" href={`mailto:${contact.email}`}>
            Get in touch
          </Anchor>
          <Anchor align="end" href={contact.website}>{contact.website}</Anchor>
        </Box>
      </Box>
    </FooterComponent>
  );
}
