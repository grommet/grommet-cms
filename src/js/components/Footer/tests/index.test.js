import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import Footer from '../index';
import props from './__mocks__/footerProps';

describe('<Footer />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Footer {...props} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
