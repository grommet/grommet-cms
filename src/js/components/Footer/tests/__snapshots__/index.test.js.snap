// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render with default props 1`] = `
<Footer
  align="center"
  colorIndex="brand"
  direction="row"
  primary={true}
  responsive={false}
>
  <Box
    announce={false}
    direction="row"
    focusable={true}
    full="horizontal"
    pad="large"
    responsive={true}
    tag="div"
  >
    <Box
      announce={false}
      basis="1/3"
      direction="column"
      focusable={true}
      justify="center"
      pad="none"
      responsive={true}
      tag="div"
    />
    <Box
      announce={false}
      basis="1/3"
      direction="column"
      focusable={true}
      justify="center"
      pad="none"
      responsive={true}
      tag="div"
    >
      <Heading
        align="center"
        className="post-feed-item--header"
        tag="h3"
      >
        Â© 2017 Grommet Labs
      </Heading>
    </Box>
    <Box
      announce={false}
      basis="1/3"
      direction="column"
      focusable={true}
      justify="center"
      pad="none"
      responsive={true}
      tag="div"
    >
      <Anchor
        align="end"
        animateIcon={true}
        href="mailto:hello@grommet.io"
        method="push"
        tag="a"
      >
        Get in touch
      </Anchor>
      <Anchor
        align="end"
        animateIcon={true}
        href="https://grommet.io"
        method="push"
        tag="a"
      >
        https://grommet.io
      </Anchor>
    </Box>
  </Box>
</Footer>
`;
